### ISSUE TITLE
support user define track label

### ISSUE THREAD

**Issue by zcabsji** [pion/ion-sfu#86]
## Summary

Hope that ion-sfu supports the user-defined label or ID of the track.

I'm new to webRTC, and I am currently using ion-sfu to develop a chat app. I find that for each track the front-end ontracked, it is hard to find which user it belongs to. I need it to support the adjustment to the user's volume. And it may be solved if I can set the label of the track as user id.

I tried to use the mapping from ssrc to label to distinguish the tracks. But the label and ID I tried to print out on backend is different from the track information I received in front-end.

If there is any other solution to this issue, please tell me as well, thanks!

Thanks for reading.

**Comment by tarrencev** [comment]
I havent seen this behavior. They should be the same. Could you share an example/test that fails?

**Comment by zcabsji** [comment]
In this test, I made three peer connection to sfu server(three front-end web page).
According to what log.Infof prints in the sfu server. The track Id should be: ba7dfc95-6293-40e2-8cc1-155841530d4e, 728da4ed-2244-4ded-8810-af53c0a206f1, and a60b8e9d-eca4-4e69-b80b-2ff58f09e6db.
![1971597473562_ pic](https://user-images.githubusercontent.com/43610574/90307050-9c664500-df05-11ea-801c-e77f5dcff696.jpg)
But when I tried to print the track information in the front-end, only a60b8e9d-eca4-4e69-b80b-2ff58f09e6db matches. Following pics are the printed tracks
![1981597473589_ pic](https://user-images.githubusercontent.com/43610574/90307178-96249880-df06-11ea-88d8-de972923ff16.jpg)
![1991597473606_ pic](https://user-images.githubusercontent.com/43610574/90307179-99b81f80-df06-11ea-9a0c-98e6a4f60400.jpg)
![2001597473629_ pic](https://user-images.githubusercontent.com/43610574/90307183-9cb31000-df06-11ea-92b1-ff49979339bf.jpg)
![2011597473659_ pic](https://user-images.githubusercontent.com/43610574/90307187-9f156a00-df06-11ea-89ee-16583e2a8337.jpg)
![2021597473667_ pic](https://user-images.githubusercontent.com/43610574/90307188-a0469700-df06-11ea-84ac-f0e7b27cce5c.jpg)

While in the SDP I received in the renegotiation offer, the track ID matches. the track ID is exactly the same as it printed in back-end:
![Screen Shot 2020-08-15 at 3 07 13 PM](https://user-images.githubusercontent.com/43610574/90307488-295ecd80-df09-11ea-838f-08f3714a2472.png)
![Screen Shot 2020-08-15 at 3 07 31 PM](https://user-images.githubusercontent.com/43610574/90307489-2a8ffa80-df09-11ea-821f-956d80803af7.png)
![Screen Shot 2020-08-15 at 3 07 53 PM](https://user-images.githubusercontent.com/43610574/90307490-2b289100-df09-11ea-950b-2f4da4e8ddba.png)

That means, SDP sent from back-end matches the track ID, but when Front-end ontracked, the track information is not the same as it in the back-end. And that makes me unable to link the user and the track.

by the way, the code for print track information is:
<img width="370" alt="WeChatcccf71f8d3859fbd880ecc0ccbbc7e5f" src="https://user-images.githubusercontent.com/43610574/90307527-91151880-df09-11ea-876b-3c919f6f52eb.png">

Thanks for reading!

**Comment by tarrencev** [comment]
@zcabsji its a little hard to follow. i'm wondering if you could modify this test to fail? https://github.com/pion/ion-sfu/blob/master/pkg/session_test.go#L160

**Comment by OrlandoCo** [comment]
@tarrencev I have been experimenting with this, and I couldn't relate the tracks with the client... the only way I found was modifying the SDP In a very hacky and ugly way or subscribing to peers one by one and not automatically to identify the tracks.

I think that the second solution can work with the benefit of saving bandwidth with selective subscribing but sometimes feels kinda slow because it takes sometime to recieve the tracks..

Or maybe the SFU shoul work as the original one with a new transport per pub/sub?

**Comment by tarrencev** [comment]
@OrlandoCo i haven't seen this issue on my end. i am using msid/track id's to maintain correspondence between user tracks and user metadata. im not sure why you are seeing a different behavior (unless i just dont understand the issue).

in my app i store a mapping of msid => MediaStream. on the server, at signaling layer, i parse SDP to get the tracks during join/negotiation and keep track of which user owns which stream. i then send the client user metadata that includes the users stream msid and use that to lookup + render the stream in the client.

**Comment by OrlandoCo** [comment]
@tarrencev I do exactly the same, keeping the same info in server, and there is not problem there, the problem is on frontend, In a very summarized way:
 1.  user **ONE** have 2 tracks with ID's **A** and **B**
 2. user **TWO** joins with 2 tracks with ID's **C** and **D**.
 3. I send info to user **ONE** that user **TWO**  joined with track ID **C** and **D**
4. User **ONE** get new track event with ID's **X** and **Y** from user **TWO** instead **C** and **D** and now I can't identify  to whom those tracks are.

**Comment by OrlandoCo** [comment]
@tarrencev Ok I finally got it, the browser changes the track id on every track event , but the stream id it's the same, I was confused because Pion add the tracks labels same as the stream id. 

So now I can relate the tracks with the users. I think the bug label can be removed.

**Comment by tarrencev** [comment]
great glad you were able to solve it! would be good to add some docs on this pattern since it seems common cool
